drivers/scale_host.go

```diff
@@ -245,10 +305,14 @@ func (s *ScaleHostDriver) Execute(conf interface{}, apiClient *client.RancherCli
 		for _, host := range hostScalingGroup {
 			state := host.State
 			if state == "inactive" || state == "deactivating" || state == "reconnecting" || state == "disconnected" {
+				if deleteCount >= amount {
+					return http.StatusBadRequest, fmt.Errorf("Cannot scale down exceed amount")
+				}
 				badHosts[host.Id] = true
 				log.Infof("Deleting host %s with priority because of bad state: %s", host.Id, host.State)
 				code, err := deleteHost(host.Id, apiClient)
 				if err != nil {
+					log.Errorf("Error in deleting host")
```

Comment:
I did a bad job explaining what I wanted here. There error you are returning here is going to ultimately be sent to the user as part of the API response. As such, it should not contain technical details; it should be simple, but meaningful. But, you should _log_ the technical details so that admins can see the logs understand what happened.

Your log message doesn't include the error with the technical details and you are instead returning it.

So to fix this, the two lines should look like:
```
log.Errorf("Error in deleting host: %v", err)
return code, fmt.Errorf("Error deleting host")
```